### PR TITLE
Updateobject role

### PR DESCRIPTION
**Pull request by pvdvreede**
This PR adds in some generic helper filters that would be useful across apps, updates the logging to output the msg and add a new role that can be used which will update a CRD object, but only if it has not changed.

There is also an increase in tests across everything!

### CODE REVIEW

File: opt/ansible-runner/ansible/roles/updateobject/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      {{
+        lookup('k8s', api_version=updateobject_apiversion, kind=updateobject_kind, resource_name=updateobject_name, namespace=updateobject_namespace)
+        | json_query('status')
+      }}
+
+- name: Checking if either statuses have changed
+  set_fact:
+    updateobject_shouldupdate_outputs: |-
+      {{
+        updateobject_current_queue_status.outputs is not defined or
```
Comment (pjmichael) [inline opt/ansible-runner/ansible/roles/updateobject/tasks/main.yml]: this is fine, but i'm just wondering if we could simplify the logic to something like:
```
not (updateobject_current_queue_status.outputs | default({}) | are_dicts_equal(updateobject_outputs))
```

Comment (pjmichael) [reply opt/ansible-runner/ansible/roles/updateobject/tasks/main.yml]: not sure if would work. was just wondering :)

Comment (pvdvreede) [reply opt/ansible-runner/ansible/roles/updateobject/tasks/main.yml]: Good idea, let me check if it will work

Comment (pvdvreede) [reply opt/ansible-runner/ansible/roles/updateobject/tasks/main.yml]: This has been updated.

**Pull request merged**